src/jit/emitxarch.cpp

```diff
@@ -8794,14 +8794,15 @@ BYTE*       emitter::emitOutputCV(BYTE* dst, instrDesc* id, size_t code, CnsVal*
         addr = emitConsBlock + doff;
 
         int byteSize = EA_SIZE_IN_BYTES(size);
-#ifdef _TARGET_AMD64_
+
+#ifndef LEGACY_BACKEND
         // this instruction has a fixed size (4) src.  
         if (ins == INS_cvttss2si || ins == INS_cvtss2sd || ins == INS_vbroadcastss)
             byteSize = 4;
         // This has a fixed size (8) source.
         if (ins == INS_vbroadcastsd)
             byteSize = 8;
-#endif
+#endif // !LEGACY_BACKEND
```

Comment:
I see quite a few such cases in emitxarch.cpp that needs to be enabled for RyuJIT x86.

Example:
emitInsCanOnlyWriteSSE2OrAVXReg()

Should we audit all #ifdef _TARGET_AMD64_ and fix in one shot?